src/Controller/Backend/ContentEditController.php

```diff
@@ -235,15 +236,15 @@ private function updateField(Content $content, string $fieldName, $value, ?strin
 
     private function updateTaxonomy(Content $content, string $key, $taxonomy): void
     {
-        $taxonomy = collect(Json::findArray($taxonomy))->filter();
+        $taxonomy = new Collection(Json::findArray($taxonomy));
 
         // Remove old ones
         foreach ($content->getTaxonomies($key) as $current) {
             $content->removeTaxonomy($current);
         }
 
         // Then (re-) add selected ones
-        foreach ($taxonomy as $slug) {
+        foreach ($taxonomy->filter() as $slug) {
```

Comment:
?

Reply:
Before: 

```
        $taxonomy = collect(Json::findArray($taxonomy))->filter();
…
        foreach ($taxonomy as $slug) {
```

after: 

```
        $taxonomy = new Collection(Json::findArray($taxonomy));
…
        foreach ($taxonomy->filter() as $slug) {
```

So the `->filter()` is in a different location, but the functionality is unaltered. All the `->filter()` does is skip over the 'empty' elements that might be in the collection.

Reply:
can you move it where it was?
I wonder why empty taxonomies are even passed to this method :thinking:

Reply:
I moved it back. :-)

Empty ones can be in there, because the Vue component _always_ passes in the value of the selection, so if the user clears an existing taxonomy, there's no selection and that means it gets one that's empty. /shrug